rust/segment/src/types.rs

```diff
@@ -889,6 +909,10 @@ pub async fn materialize_logs(
         res.push(value);
     }
     res.sort_by(|x, y| x.offset_id.cmp(&y.offset_id));
+
+    tracing::info!("Total length of materialized logs: {}", res.len());
```

Comment:
nit: can combine this trace and the above trace - `tracing::info!("Total length of logs in materializer: {}", logs.total_len())`?